### PR TITLE
feat(efb): fuel page

### PR DESCRIPTION
**Pull request by Inexorable95**
## Summary of Changes
The user will be able to monitor the fuel quantity and refuel/defuel the A320 from the EFB's fuel page 

- [x] Display current fuel quantity
- [x] Set a desired fuel quantity
- [x] Check if refueling is possible (plane is not moving, on the ground, and both engines are off) 
- [x] Refueling is only possible when either BUS HOT 1 OR BUS DC 2 are powered. (in practice, it means refueling always works, unless batteries are depleted)
- [x] Implement airplane SVG
- [x] Start/Stop the refuel process (plane is on the ground, with a ground speed of 0 and both engines off and refuel button is pressed)
- [x] Implement user-chosen refuel speeds (real, fast or instantaneous)
- [x] Wire all values to the sim
- [x] Handle different units (kgs/lbs) (Logic implemented, but right now persistence vars are not implemented in the EFB) **This means switching between kg and lb will not work currently, but will in the future** 
- [x] ~~Call the fuel truck from the fuel page~~ This has been done and tested, but when the truck arrives, it automatically opens the MFS refuel interface (we don't need it), if the user closes the interface the truck goes away. Relying on the truck being present for refueling would be detrimental to the user experience (it will confuse users who may use the MFS refueling interface instead of the EFB one). Hopefully we can soon call the truck and have it stay without popping up the MFS interface, until then it's best to keep this as is.

## Screenshots (if necessary)
refuelTest2

## References
ref1
ref2
ref3

## Additional context

Discord username (if different from GitHub): Inexorable

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Comment by Benjozork** [comment]
#3656 introduced new slider components, wanna use those now. eventually, we should switch to `@flybywiresim/react-components`

**Comment by beheh** [comment]
I'm not sure if you can control this at this time, but opening the "real" refueling panel should reset this ECAM messages as the valves close:

https://github.com/flybywiresim/a32nx/blob/a2719fabe0557d46dc458b8b2d8771d40aa00912/A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js#L1171-L1179

**Comment by davideguadagno** [comment]
Will easily test this beauty of a code  <3

**Comment by SaqibSiddiqi** [comment]
If you need the code for realistic refueling to refer to #3466 PR. I have the values right for the current Fuel Flow to match the real-world refueling timing. However, it likes to overflow the main tank. Hope it helps and you can fix the issues I had!

**Comment by Inexorable95** [comment]
> If you need the code for realistic refueling to refer to #3466 PR. I have the values right for the current Fuel Flow to match the real-world refueling timing. However, it likes to overflow the main tank. Hope it helps and you can fix the issues I had!

I'll look into it, thanks for the info!

**Comment by Inexorable95** [comment]
Ready to be tested

File: .husky/.gitignore [hunk 1]
```diff
@@ -1 +1 @@
-_
+_
```
Comment (Benjozork) [inline .husky/.gitignore:1]: Can you revert this?

File: A32NX/html_ui/Pages/A32NX_Core/A32NX_Refuel.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
+const REFUEL_FACTOR = 0.42;
+const CENTER_MODIFIER = 3;
+
+class A32NX_Refuel {
+    constructor() {}
+
+    init() {
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/A32NX_Core/A32NX_Refuel.js:7]: Please add the `console.log` messages printed on initialization/init like other core modules

File: package.json [hunk 3]
```diff
@@ -69,6 +70,7 @@
     "lodash": "^4.17.20",
     "react": "^17.0.0",
     "react-dom": "^17.0.0",
+    "react-input-slider": "^6.0.0",
```
Comment (Benjozork) [inline package.json:73]: We have a slider component in `@flybywiresim/react-components`, can you try using that ?

File: src/instruments/src/EFB/Components/Form/SimpleInput/SimpleInput.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    placeholder?: string,
+    value?: any,
+    onChange?: (value: string) => void,
+    min?: number,
+    max?: number,
+    number?: boolean,
+    noLeftMargin?: boolean,
+    reverse?: boolean // Flip label/input order,
+    className?: string
+};
```
Comment (Benjozork) [inline src/instruments/src/EFB/Components/Form/SimpleInput/SimpleInput.tsx:14]:
```suggestion
}
```

File: src/instruments/src/EFB/Components/Form/SimpleInput/SimpleInput.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+type SimpleInputState = {
+    value: string
+};
```
Comment (Benjozork) [inline src/instruments/src/EFB/Components/Form/SimpleInput/SimpleInput.tsx:18]:
```suggestion
}
```

File: src/instruments/src/EFB/Components/Form/SimpleInput/SimpleInput.tsx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+export default class SimpleInput extends React.Component<SimpleInputProps, SimpleInputState> {
```
Comment (Benjozork) [inline src/instruments/src/EFB/Components/Form/SimpleInput/SimpleInput.tsx:20]: Please prefer function components

File: src/instruments/src/EFB/Components/Progress/Progress.scss [hunk 7]
```diff
@@ -0,0 +1,12 @@
+.vertical-progress-bar {
+  @apply bg-blue-light-contrast;
+  position: absolute;
+  margin-top: -3px;
+  width: 5px;
+  height: 31px;
+  // background-color: #5e72e4;
```
Comment (Benjozork) [inline src/instruments/src/EFB/Components/Progress/Progress.scss:7]: Remove commented code

File: src/instruments/src/EFB/Components/Progress/Progress.tsx [hunk 8]
```diff
@@ -0,0 +1,150 @@
+import * as React from "react";
```
Comment (Benjozork) [inline src/instruments/src/EFB/Components/Progress/Progress.tsx:1]: Use 4 space indents

File: src/instruments/src/EFB/Components/Progress/Progress.tsx [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+  width: "100%",
+  borderRadius: "50px",
+  labelAlignment: "right",
+  baseBgColor: "#e0e0de",
+  labelColor: "#fff",
+  labelSize: "15px",
+  isLabelVisible: true,
+  displayBar:false,
+  completedBar: 0
+};
```
Comment (Benjozork) [inline src/instruments/src/EFB/Components/Progress/Progress.tsx:148]: Do not use `propTypes` or `defaultProps` as we use TypeScript

File: src/instruments/src/EFB/Components/Progress/Progress.tsx [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+
+export default ProgressBar;
```
Comment (Benjozork) [inline src/instruments/src/EFB/Components/Progress/Progress.tsx:150]: Don't use default exports

File: src/instruments/src/EFB/Dashboard/Dashboard.tsx [hunk 11]
```diff
@@ -41,12 +41,13 @@ type DashboardProps = {
 }
 
 const Dashboard: React.FC<DashboardProps> = (props) => {
-    const [flightNumber] = useSimVar('ATC FLIGHT NUMBER', 'String', 1_000);
-    const [aircraftType] = useSimVar('TITLE', 'String', 1_000);
-    const [altitude] = useSimVar('PLANE ALTITUDE', 'feet', 500);
-    const [heading] = useSimVar('PLANE HEADING DEGREES TRUE', 'degrees', 500);
-    const [latitude] = useSimVar('PLANE LATITUDE', 'degree latitude', 500);
-    const [longitude] = useSimVar('PLANE LONGITUDE', 'degree longitude', 500);
+    //TODO: remove those false variables when we merge
```
Comment (Benjozork) [inline src/instruments/src/EFB/Dashboard/Dashboard.tsx:44]:
```suggestion
    // TODO: remove those false variables when we merge
```

File: src/instruments/src/EFB/Dispatch/Pages/FuelPage.tsx [hunk 12]
```diff
@@ -0,0 +1,281 @@
...
+            </div>
+            <div className='wrapper hiding-block hiding-block-top-left'></div>
+            <div className='wrapper hiding-block hiding-block-bottom-left'></div>
+            <div className='wrapper hiding-block hiding-block-top-right'></div>
+            <div className='wrapper hiding-block hiding-block-bottom-right'></div>
+        </div>
+    );
+};
+
+export default FuelWidget;
```
Comment (Benjozork) [inline src/instruments/src/EFB/Dispatch/Pages/FuelPage.tsx:281]: Don't use default exports

**Review by Benjozork — CHANGES_REQUESTED** [review]
Few style tweaks but overall LGTM. I will review the UI a bit later.

You also have quite a few linting issues to fix, didn't point out the formatting errors because those are caught already by the lint.

File: package.json (hunk 3 above)
Comment (Inexorable95) [reply package.json:73]: I am, this was an old component I used during my initial tests, i'll remove it from the package

File: src/instruments/src/EFB/Components/Form/SimpleInput/SimpleInput.tsx [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+    placeholder?: string,
+    value?: any,
+    onChange?: (value: string) => void,
+    min?: number,
+    max?: number,
+    number?: boolean,
+    noLeftMargin?: boolean,
+    reverse?: boolean // Flip label/input order,
+    className?: string
+};
```
Comment (Benjozork) [inline src/instruments/src/EFB/Components/Form/SimpleInput/SimpleInput.tsx:14]: Not resolved

File: src/instruments/src/EFB/Components/Form/SimpleInput/SimpleInput.tsx [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+
+type SimpleInputState = {
+    value: string
+};
```
Comment (Benjozork) [inline src/instruments/src/EFB/Components/Form/SimpleInput/SimpleInput.tsx:18]: Not resolved

**Review by Benjozork — CHANGES_REQUESTED** [review]

**Comment by St54Kevin** [comment]
What does the Refuel number field input show? The amount that is being refueled or the amount i want to be in the tank after refueling? IRL we usually tell how much fuel we want in total and the tanker will care of how much that is to refuel. Even on the refueling panel in the cockpit we set the total fuel wanted. Just wanted to check how you have currently it implemented.

**Comment by Inexorable95** [comment]
> What does the Refuel number field input show? The amount that is being refueled or the amount i want to be in the tank after refueling? IRL we usually tell how much fuel we want in total and the tanker will care of how much that is to refuel. Even on the refueling panel in the cockpit we set the total fuel wanted. Just wanted to check how you have currently it implemented.

The input field is where you input the total amount of fuel you want to have, yes.
then the defuel/refuel will happen automatically is the refuel conditions are met (both engines off, ground speed of 0 and the plane is on the ground)

File: src/instruments/src/EFB/Components/Form/SimpleInput/SimpleInput.tsx (hunk 13 above)
Comment (Inexorable95) [reply src/instruments/src/EFB/Components/Form/SimpleInput/SimpleInput.tsx:14]: Will be resolved by the original creator of the Landing performance PR

File: src/instruments/src/EFB/Components/Form/SimpleInput/SimpleInput.tsx (hunk 14 above)
Comment (Inexorable95) [reply src/instruments/src/EFB/Components/Form/SimpleInput/SimpleInput.tsx:18]: Will be resolved by the original creator of the Landing performance PR

**Review by Inexorable95 — COMMENTED** [review]
fixed

**Review by Inexorable95 — COMMENTED** [review]
fixed

**Comment by St54Kevin** [comment]
Category: QA Tester
Name: St54Kevin#3583
Date of testing: 18/03/2021
Version of the sim: 1.14.5.0
PR Tested: #3744 
Tier of testing: 2
Changes to observe: EFB fuel page
Steps you've made to test the Pull Request:

1. Loaded up C&D
2. Tested refueling/defueling via the EFB

Media:

Issues (If Any): 
- Fuel is instantly changing when putting in a number into the refuel field. That causes the fuel to drop to 0kg when deleting the value to type in a new one. That could be a problem with APU running or even an engine running. I suggest having a button to START refueling after the value input.
- The vertical cyan bars on each tank bars gave me the impression that Im able to change them individually, but they arent. I suggest removing them as I dont see a reason to be able to set each individual tank, as there is a system to it anyways.
- The "kgs" in the refuel field is a little far to the right. I would just delete the "s" as a "kgs" is rarely used ever to indicate multiple kilogramms 
image
- The blue bar does not match with the cyan bar after refueling is finished 
image
- Center tank shows 1kg in it when it should be empty. When setting 0kg fuel overall every tank shows 0kg except the center tank
- The blue "defueling" and "refueling" and green "Available" is a little small for my liking. I would make it more prominent to really show that there is something happening. But that could also be done with the above START button which could change color or naming.

Overall Rating: 

Conclusions of the testing: Needs some cosmetic and usability improvements. Issue #1 personally is mandatory for me to feel good to release it. Overall the page looks really nice and blends perfect into FBW design

**Comment by Inexorable95** [comment]
> Fuel is instantly changing when putting in a number into the refuel field. That causes the fuel to drop to 0kg when deleting the value to type in a new one. That could be a problem with APU running or even an engine running. I suggest having a button to START refueling after the value input.

It could indeed cause issues with a running APU, however the refuel/defuel will not happen if an engine is running, I'll add a start/stop button for clarity.

> Units did not change after changing over to lbs, although it did correctly refuel, but all was in kg on the EFB

Units are currently implemented but not supported because synchronisation with NXDataStore vars (what is used in the MCDU) is not currently supported, it will be in the future, so the code is there for the future, but right now it doesn't work.
> The vertical cyan bars on each tank bars gave me the impression that Im able to change them individually, but they arent. I suggest removing them as I dont see a reason to be able to set each individual tank, as there is a system to it anyways.

The cyan bars are there to indicate how each tank will be filled when the refuel is completed, they may be useless when using instant refueling, but when using real refueling (which can take up to 20 minutes) they give important information. I see what you mean about them looking like they can be changed, that's because they use the same color as the slider, I'll change them to grey to avoid this issue.
> * The "kgs" in the refuel field is a little far to the right and a little small. I would just delete the "s" as a "kgs" is rarely used ever to indicate multiple kilogramms. 

True, i'll remove the S, but the size and placement is coherent with other "kg" labels on the page.

>   image
> * The blue bar does not match with the cyan bar after refueling is finished
>   image

That's a matter of preference, the bar touches the slider, its left segment touches the end of the bar, indicating that it is the limit, that's how the component works and I can tweak the settings a bit, but it looks as it should look as far as I'm concerned

> * Center tank shows 1kg in it when it should be empty. When setting 0kg fuel overall every tank shows 0kg except the center tank

bug, will check it out

> * The blue "defueling" and "refueling" and green "Available" is a little small for my liking. I would make it more prominent to really show that there is something happening. But that could also be done with the above START button which could change color or naming.

I can make it a bit bigger, but with the button we'll have more info indeed

I'll keep you updated when all of this is pushed. Thanks

**Comment by St54Kevin** [comment]
A suggestion for the button: It could be named "REFUEL" and "DEFUEL" wether a lower or high fuel value is set. Then it could pulse blue or amber when its operating and turning green again when finished. BTW would a small time frame for the progress be cool? Like when I set 10000kg it shows me an estimate of the duration at the bottom for each option for the process to complete depending on current fuel in the plane?

**Comment by Inexorable95** [comment]
> A suggestion for the button: It could be named "REFUEL" and "DEFUEL" wether a lower or high fuel value is set. Then it could pulse blue or amber when its operating and turning green again when finished. BTW would a small time frame for the progress be cool? Like when I set 10000kg it shows me an estimate of the duration like "~2.5min" at the bottom for each option for the process to complete depending on current fuel in the plane?

Yeah I had the same idea for the button, but I think Start and Stop is more clear. as for the colors, I will adapt the color of the button according to whether it's refueling/defueling/available/unavailable.
As for the estimated time, it's pretty quick to do, i'll add it below the button

**Comment by Inexorable95** [comment]
all recommandations added, testing can be resumed.

**Comment by marcman86** [comment]
Category: QA Tester
Name: marcman86#4907
Date of testing: 18.3.2021
Version of the sim: 1.14.5.0
PR Tested: #3744
Tier of testing: 1
Changes to observe: EFB fuel page
Testing technique: Tryout

Steps made to test the Pull Request:

1.) Spawned C&D at EDDP
2.) Turned on bat., ext. power and EFB
3.) Played with fuel loader

Issues: 
- AC is set to "kg", Fuel page still uses Lbs
- Lbs is not a unit from my knowledge. Please use lb and kg (in reference images it shows Kgs). Alternative is to match it to the MCDU KG and LBS.
- The refuel value field (dark blue) is cut off after the Lbs 

Design changes i'd like to see:

- Refuel field is in the sub-tab line, needs to be shifted down a bit.
- Fuel values of the tanks are tiny, wasted space
- Tank names should be centered
- Est. Time should be centered under the button and larger

I did a paint image from my ideas:



Overall Rating: -

Conclusions of the testing: Very nice PR! Works fine. But some design changes would be nice to see.

Hope this helps :-)

**Comment by Inexorable95** [comment]
> Category: QA Tester
> Name: marcman86#4907
> Date of testing: 18.3.2021
> Version of the sim: 1.14.5.0
> PR Tested: #3744
> Tier of testing: 1
> Changes to observe: EFB fuel page
> Testing technique: Tryout
> 
> Steps made to test the Pull Request:
> 
> 1.) Spawned C&D at EDDP
> 2.) Turned on bat., ext. power and EFB
> 3.) Played with fuel loader
> 
> Issues:
> 
> * AC is set to "kg", Fuel page still uses Lbs
> * Lbs is not a unit from my knowledge. Please use lb and kg (in reference images it shows Kgs). Alternative is to match it to the MCDU KG and LBS.
> * The refuel value field (dark blue) is cut off after the Lbs
> 
> Design changes i'd like to see:
> 
> * Refuel field is in the sub-tab line, needs to be shifted down a bit.
> * Fuel values of the tanks are tiny, wasted space
> * Tank names should be centered
> * Est. Time should be centered under the button and larger
> 
> I did a paint image from my ideas:
> 
> 
> 
> Overall Rating: -
> 
> Conclusions of the testing: Very nice PR! Works fine. But some design changes would be nice to see.
> 
> Hope this helps :-)

Thanks for the review, please note that the unit switch (changing between KG and LB) is not supported right now, as persistance is not supported yet, I mentionned it in the PR presentation.

I changed Lbs to LB and Kg to KG in the latest version

As for the design suggestion, I'll see what I can do to make the text bigger, but the tank names being on the left was a design choice.

**Comment by Inexorable95** [comment]
labels are now bigger, and less useless blank space is present

**Comment by marcman86** [comment]
Edit of:
> Category: QA Tester
> Name: marcman86#4907
> Date of testing: 18.3.2021
> Version of the sim: 1.14.5.0
> PR Tested: #3744
> Tier of testing: 1
> Changes to observe: EFB fuel page
> Testing technique: Tryout

Retested:

- Every issue from mylast test is fixed
- Also tested that tanking is inhibited when leaving parking space

Overall Rating: Good

Conclusions of the testing: Good to go

Hope this helps :-)

**Comment by St54Kevin** [comment]
Category: QA Tester
Name: St54Kevin#3583
Date of testing: 19/03/2021
Version of the sim: 1.14.5.0
PR Tested: #3744 
Tier of testing: 1
Changes to observe: EFB Fuel Page

**Steps you've made to test the Pull Request:**
1. Loaded up C&D
2. Played with EFB fuel page
3. Tested all above issue cases

**Media:**

**Issues (If Any):** 
- Fueling process is possible without BAT power. As stated in the PRs reference doc, it shouldnt. Sry, didnt test this before.
- If we wanna be correct, tank capacities are a bit off, although only some kg. We need +1kg on each wing tank and -4 on the center (I added the reference doc values, they do not sum to 18979 WTF :D)

**Conclusions of the testing:** LGTM overall, very minor issues. All major ones above are fixed or improved. I like it

**Comment by Matse08** [comment]
Category: QA Tester
Name: Matse#7454
Date of testing: 19.03.21
Version of the sim: 1.14.5.0
PR Tested: #3744
Tier of testing: 2
Changes to observe:New fuel page in EFB


Testing technique: Try-out
Steps you've made to test the Pull Request:
1. Spawned cold and dark
2. Turned on efb 
3. filled the tanks to max 


Media: 
Microsoft Flight Simulator 19 03 2021 14_08_07

Issues (If Any): 
After starting engines it says that i cant change the fuel
Steps to achieve the issue:
Start engine

Overall Rating: To be improved
Conclusions of the testing: There should be a way to slide all sliders.

**Comment by myselfaaryan** [comment]
> Category: QA Tester
> Name: Matse#7454
> Date of testing: 19.03.21
> Version of the sim: 1.14.5.0
> PR Tested: #3744
> Tier of testing: 2
> Changes to observe:New fuel page in EFB
> 
> Testing technique: Try-out
> Steps you've made to test the Pull Request:
> 
> 1. Spawned cold and dark
> 2. Turned on efb
> 3. filled the tanks to max
> 
> Media:
> Microsoft Flight Simulator 19 03 2021 14_08_07
> 
> Issues (If Any):
> After starting engines it says that i cant change the fuel
> Steps to achieve the issue:
> Start engine
> 
> Overall Rating: To be improved
> Conclusions of the testing: There should be a way to slide all sliders.

Disabled refueling after engine start is intentional.

**Comment by Inexorable95** [comment]
> Category: QA Tester
> Name: Matse#7454
> Date of testing: 19.03.21
> Version of the sim: 1.14.5.0
> PR Tested: #3744
> Tier of testing: 2
> Changes to observe:New fuel page in EFB
> 
> Testing technique: Try-out
> Steps you've made to test the Pull Request:
> 
> 1. Spawned cold and dark
> 2. Turned on efb
> 3. filled the tanks to max
> 
> Media:
> Microsoft Flight Simulator 19 03 2021 14_08_07
> 
> Issues (If Any):
> After starting engines it says that i cant change the fuel
> Steps to achieve the issue:
> Start engine
> 
> Overall Rating: To be improved
> Conclusions of the testing: There should be a way to slide all sliders.

in real life you never pick how fuel is set in the tanks, it always starts with the wing tanks (first outer, than inner) and finally the center tank, therefore it is not planned to be able to slide all the sliders

as for the refueling being disabled when the engines are on, that is normal and intentional

**Comment by Inexorable95** [comment]
> Category: QA Tester
> Name: St54Kevin#3583
> Date of testing: 19/03/2021
> Version of the sim: 1.14.5.0
> PR Tested: #3744
> Tier of testing: 1
> Changes to observe: EFB Fuel Page
> 
> **Steps you've made to test the Pull Request:**
> 
> 1. Loaded up C&D
> 2. Played with EFB fuel page
> 3. Tested all above issue cases
> 
> **Media:**
> 
> **Issues (If Any):**
> 
> * Fueling process is possible without BAT power. As stated in the PRs reference doc, it shouldnt. Sry, didnt test this before.
> * If we wanna be correct, tank capacities are a bit off, although only some kg. We need +1kg on each wing tank and -4 on the center (I added the reference doc values, they do not sum to 18979 WTF :D)
> 
> **Conclusions of the testing:** LGTM overall, very minor issues. All major ones above are fixed or improved. I like it

MFS has unusable fuel in each tanks, I guess it is a way to simulated the vent tanks, I don't know. Either way, those small difference are expected for the moment.

concerning the refueling without power, i'll look into it

**Comment by Inexorable95** [comment]
Refueling now requires DC power (APU or BATT or GPU) (PR updated)

**Comment by easyjetsimpilot** [comment]
Category: QA Tester
Name: Gavin (Easyjetsimpilot) 
Date of testing: 21.3.2021
Version of the sim: 1.14.5.0
PR Tested: #3744
Tier of testing: 2
Changes to observe: EFB fuel page
Testing technique: Tryout

Steps made to test the Pull Request:

1.) Spawned C&D and emptied fuel tanks using MSFS Fuel window 
2.) re-loaded 5 tonne of fuel for a flight as requested by OFP using the EFB. Excellent refuelling times (approx 1000kg per min).  Checked against fuel ECAM pages. Matched perfectly well. 
3.) Watched fuel being used throughout the flight both on ECAM and Flypad.

Possible Improvements: Would be nice if you can call the fuel truck and it remain throughout refueling. 

The aircraft always spawns with 9.5 tonnes on board. Would be nice if it spawned with a more realistic fuel amount on board (1.8-3.0 tonnes?) for after a flight. 


Overall Rating: - Excellent working of the refuelling system. Some changes if possible would really help it along. But happy with how it works for now.

Hope this helps.

**Comment by St54Kevin** [comment]
Category: QA Tester
Name: St54Kevin#3583
Date of testing: 21/03/2021
Version of the sim: 1.14.5.0
PR Tested: #3744 
Tier of testing: 1
Changes to observe: EFB Fuel Page

**Steps you've made to test the Pull Request:**
1. Quick check of the fuel page function on the stand
2. 
**Media:**

**Issues (If Any):** - 

**Conclusions of the testing:** LGTM. no problems

**Review by Benjozork — APPROVED** [review]

**Pull request merged**